### PR TITLE
Reboot after attaching EBS volumes.

### PR DESCRIPTION
**Pull request by rajachan**
Adding a reboot that subscribes to the execute action which attaches EBS volumes to the instance. This is necessary so the outdated NVMe driver in CentOS 6 can discover the newly added volume.

### CODE REVIEW

File: recipes/_master_base_config.rb [hunk 1]
```diff
@@ -33,12 +33,18 @@
...
+  subscribes :reboot_now, "execute[attach_volume]", :immediately
+end
+
 # wait for the drive to attach, before making a filesystem
 ruby_block "sleeping_for_volume" do
   block do
     wait_for_block_dev(dev_path)
   end
-  subscribes :run, "execute[attach_volume]", :immediately
+  subscribes :run, "reboot[ec2_nvme_reboot]", :immediately
```
Comment (bwbarrett) [inline recipes/_master_base_config.rb]: why would we do this all this time?  Seems better to only do it with NVMe volumes.  Even then, not sure that I love the reboot here.  Shouldn't the volume be attached when the instance launches with the cloudformation template?

**Comment by dougalb** [comment]
Maybe no support for NVME devices or instances in CentOS 6? At it's core it is now over 7 years old - https://en.wikipedia.org/wiki/Red_Hat_Enterprise_Linux#RHEL_6

**Comment by bwbarrett** [comment]
@dougalb, that was where I was going...  I wasn't ok leaving CentOS 7 behind, but it really feels like leaving CentOS 6 behind is the right choice, given all the weird corner case issues.

**Comment by rajachan** [comment]
@bwbarrett  and I discussed this offline and decided to add a conditional reboot for CentOS 6 for now. I will also check to see if triggering a rescan of the PCI bus helps the driver discover the volumes; that might be a cleaner solution. This seems to be the final hurdle to get a CentOS 6 MasterServer running on C5s, so doing something conditionally for this might be worthwhile.

**Comment by rajachan** [comment]
Well, a rescan of the the PCI bus with `echo 1 > /sys/bus/pci/rescan`works. That's pretty much what we need then, instead of the reboot.

**Pull request closed without merging**